TrafficSimulator/Assets/Utilities/CarSpawner.cs

```diff
@@ -58,39 +62,130 @@ public class CarSpawner : MonoBehaviour
         private List<GameObject> _vehicleTypes;
 
         private List<Lane> _lanes = new List<Lane>(); 
-        private List<float> _lengths = new List<float>();
-        private List<float> _ratios = new List<float>();
         private List<int> _indexes = new List<int>();
-        private List<int> _maxCarsPerLane = new List<int>();
 
         private LaneNode _laneNodeCurrent;
 
         private GameObject _currentCar;
 
         private int _carCounter = 0;
-        private float _offset = 0;
         private float _carLength;
 
         private bool _spawned = false;
-        
-        private void Start()
+
+        private List<CarsForLane> spawnableLanes = new List<CarsForLane>();
+
+        // Create a empty gameobject to hold all cars
+        private GameObject _carContainer;
+
+        public struct SpawnableLaneNode
+        {
+            private float _distance;
+            private LaneNode _node;
+
+            public float Distance { get => _distance; set => _distance = value; }
+            public LaneNode Node { get => _node; set => _node = value; }
+
+            public SpawnableLaneNode(float distance, LaneNode node)
+            {
+                _distance = distance;
+                _node = node;
+            }
+        }
+
+        public class CarsForLane
+        {
+            private Lane _lane;
+            public int CapacityLeft;
+            private List<SpawnableLaneNode> _spawnableLaneNodes;
+
+            public Lane Lane { get => _lane; }
+            public List<SpawnableLaneNode> SpawnableLaneNodes { get => _spawnableLaneNodes; }
+            public int Capacity { get => _spawnableLaneNodes.Count; }
+
+            public CarsForLane(Lane lane, List<SpawnableLaneNode> spawnableLaneNodes)
+            {
+                _lane = lane;
+                _spawnableLaneNodes = spawnableLaneNodes;
+                CapacityLeft = Capacity;
+            }
+        }
+
+        void Awake()
+        {
+            _carContainer = new GameObject("CarContainer");
```

Comment:
Needs to be a check that this GameObject does not already exist due to serialisation issues